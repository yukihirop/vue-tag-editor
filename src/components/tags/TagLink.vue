<template>
  <span>
    <span
      ref="tagname"
      @click="emitClickTag"
    >
      <a>
        <span>{{ tagname }}</span>
      </a>
    </span>
    <tag-delete-button
      :tagname="tagname"
      :event-hub="eventHub"
      @delete-tag="emitDeleteTag"
    />
  </span>
</template>

<script>
import TagDeleteButton from './TagDeleteButton'

export default {
  name: "TagLink",
  components: {
    TagDeleteButton: TagDeleteButton
  },
  props:{
    tagname:{
      type: String,
      default: ''
    },
    eventHub: {
      type: Object,
      default(){
        return null
      }
    }
  },
  methods: {
    emitDeleteTag(){
      this.$emit('delete-tag')
    },
    emitClickTag(){
      this.eventHub.$emit('click-tag', this.$refs.tagname.textContent)
    }
  }
}
</script>

<style scoped="true">
a {
  text-decoration: none;
}
a:hover {
  text-decoration: underline;
  cursor: pointer;
}
</style>