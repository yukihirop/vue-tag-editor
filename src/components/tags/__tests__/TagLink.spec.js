import { createLocalVue, shallowMount } from '@vue/test-utils'
import TagLink from '../TagLink'

const localVue = createLocalVue()

const wrapperEventHub = shallowMount(TagLink, {
  localVue,
  propsData: {
    tagname: 'test'
  }
})
const eventHub = wrapperEventHub.vm

describe('TagLink', () => {
  const wrapper = shallowMount(TagLink, {
    localVue,
    propsData: {
      tagname: 'test',
      eventHub: eventHub
    }
  })

  it('correct name options', () => {
    expect(wrapper.vm.$options.name).toBe('TagLink')
  })

  it('renders correctly (snapshots)', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the correct markup', () => {
    expect(wrapper.html()).toContain("<span><span><a><span>test</span></a></span>")
    expect(wrapper.html()).toContain("tag-delete-button-stub")
  })

  it('has a delete button', () => {
    expect(wrapper.contains('tag-delete-button-stub')).toBe(true)
  })

  it('delete button click should call emitDeleteTag', () => {
    wrapper.vm.emitDeleteTag()

    expect(wrapper.emitted('delete-tag')).toBeTruthy()
    expect(wrapper.emitted('delete-tag').length).toBe(1)
  })

  it('tag-click button click should call emitTagClick', () => {
    const anchor = wrapper.find({ ref: 'tagname'})
    anchor.trigger('click')

    expect(wrapperEventHub.emitted('click-tag')).toBeTruthy()
    expect(wrapperEventHub.emitted('click-tag').length).toBe(1)
    expect(wrapperEventHub.emitted('click-tag')[0]).toEqual(['test'])
  })
})