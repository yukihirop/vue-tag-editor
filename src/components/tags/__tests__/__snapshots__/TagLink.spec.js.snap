// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagLink renders correctly (snapshots) 1`] = `
<span>
  <span>
    <a>
      <span>
        test
      </span>
    </a>
  </span>
   
  <tag-delete-button-stub
    eventhub="[object Object]"
    tagname="test"
  />
</span>
`;
